### ISSUE TITLE
tempfiles appearing in working directory

### ISSUE THREAD

**Issue by skjerns** [spyder-ide/spyder#12940]
### What steps will reproduce the problem?

Since a few weeks I'm having a strange bug that tempfiles like `tmp0i9oim8x` are appearing in the working directory of scripts. They contain a copy of unsaved code.

This happens like that:

1. Have a main script that loads another script and executes a function in it
2. Change code in the second script, but don't save the file
3. Run main script.

A copy of the changes of the second script can be found in the working directory inside a tempfile. However, this strangely doesn't happen all the time. 

I already reset Spyder, reinstalled Anaconda, deleted the .spyder3-py folder, removed all python related config folders in `~`, no changes. 

I assume it has something to do with the on-the-fly reloading of scripts. However, I could not find any `tempfile` reference in the Spyder source code that could be the culprit. Any idea why or where this happens?

## Versions

* Spyder version: 4.1.3 
* Python version: 3.7.7
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0           :  1.4.0 (OK)
chardet >=2.0.0                :  3.0.4 (OK)
cloudpickle >=0.5.0            :  1.4.1 (OK)
diff_match_patch >=20181111    :  20181111 (OK)
intervaltree                   :  None (OK)
IPython >=4.0                  :  7.13.0 (OK)
jedi =0.15.2                   :  0.15.2 (OK)
nbconvert >=4.0                :  5.6.1 (OK)
numpydoc >=0.6.0               :  0.9.2 (OK)
paramiko >=2.4.0               :  2.7.1 (OK)
parso =0.5.2                   :  0.5.2 (OK)
pexpect >=4.4.0                :  4.8.0 (OK)
pickleshare >=0.4              :  0.7.5 (OK)
psutil >=5.3                   :  5.7.0 (OK)
pygments >=2.0                 :  2.6.1 (OK)
pylint >=0.25                  :  2.5.2 (OK)
pyls >=0.31.9;<0.32.0          :  0.31.10 (OK)
qdarkstyle >=2.8               :  2.8.1 (OK)
qtawesome >=0.5.7              :  0.7.0 (OK)
qtconsole >=4.6.0              :  4.7.4 (OK)
qtpy >=1.5.0                   :  1.9.0 (OK)
rtree >=0.8.3                  :  0.9.4 (OK)
sphinx >=0.6.6                 :  3.0.3 (OK)
spyder_kernels >=1.9.1;<1.10.0 :  1.9.1 (OK)
watchdog                       :  None (OK)
zmq >=17                       :  18.1.1 (OK)

# Optional:
cython >=0.21                  :  0.29.17 (OK)
matplotlib >=2.0.0             :  3.1.3 (OK)
numpy >=1.7                    :  1.18.1 (OK)
pandas >=0.13.1                :  1.0.3 (OK)
scipy >=0.17.0                 :  1.4.1 (OK)
sympy >=0.7.3                  :  1.5.1 (OK)
```

**Comment by dalthviz** [comment]
Hi @skjerns thanks for the feedback! We currently have a logic for autosaving files at some scheduled rate. Could you try to disable the autosave option and check if that stops the creation of temp files?

You can found the option in the `Preferences` dialog here:
![image](https://user-images.githubusercontent.com/16781833/83658033-4a10bb80-a587-11ea-987c-1104a60282e9.png)

Thanks!

**Comment by skjerns** [comment]
@dalthviz I disabled the option but the files still appear. Any other ideas?

**Comment by dalthviz** [comment]
Thanks for the confirmation @skjerns although I wasn't able to reproduce this issue either with the steps provided :/

In the main time, could you help us out providing a reproducible example, please? (a simple example with some snippets that consistently causes the creation of the tempfiles). Thanks!

**Comment by skjerns** [comment]
It seems like I cannot reproduce the issue reliably myself.

However, the files keep appearing, after a working session I have several dozen of them in my working directory.

Any hints would be appreciated. It clearly appears as soon as I press "run", but strangely not always.

**Comment by ccordoba12** [comment]
@skjerns, could you share a simple code example that generates the issue for you? That way we could try to reproduce it in our side.

**Comment by skjerns** [comment]
Unfortunately I cannot reproduce an example that reliably reproduces this bug, it only happens once in a while. However, I have hooked into `tempfile` and reproduced the following traceback whenever a file appears. This traceback appears hundreds of times during one run of my script, however, only once in a while the tempfile does not get deleted properly and the tempfile remains.

```
"C:\Users\skjerns\anaconda3\Scripts\spyder-script.py", line 10, in <module>
    sys.exit(main())
  File "C:\Users\skjerns\anaconda3\lib\site-packages\spyder\app\start.py", line 205, in main
    mainwindow.main()
  File "C:\Users\skjerns\anaconda3\lib\site-packages\spyder\app\mainwindow.py", line 3748, in main
    mainwindow = run_spyder(app, options, args)
  File "C:\Users\skjerns\anaconda3\lib\site-packages\spyder\app\mainwindow.py", line 3664, in run_spyder
    app.exec_()
  File "C:\Users\skjerns\anaconda3\lib\site-packages\spyder\plugins\editor\widgets\editor.py", line 1967, in save
    self._write_to_file(finfo, finfo.filename)
  File "C:\Users\skjerns\anaconda3\lib\site-packages\spyder\plugins\editor\widgets\editor.py", line 1923, in _write_to_file
    fileinfo.encoding = encoding.write(txt, filename, fileinfo.encoding)
  File "C:\Users\skjerns\anaconda3\lib\site-packages\spyder\utils\encoding.py", line 268, in write
    mode=mode) as textfile:
  File "C:\Users\skjerns\anaconda3\lib\contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "C:\Users\skjerns\anaconda3\lib\site-packages\atomicwrites\__init__.py", line 166, in _open
    with get_fileobject(**self._open_kwargs) as f:
  File "C:\Users\skjerns\anaconda3\lib\site-packages\atomicwrites\__init__.py", line 184, in get_fileobject
    dir=dir)
  File "C:\Users\skjerns\anaconda3\lib\tempfile.py", line 355, in mkstemp
    return _mkstemp_inner(dir, prefix, suffix, flags, output_type)
  File "C:\Users\skjerns\anaconda3\lib\tempfile.py", line 269, in _mkstemp_inner
    f.writelines(traceback.format_stack(limit=15))
```

Is this of any help? Why are there tempfiles created at all?

**Comment by ccordoba12** [comment]
> edit2: seems to be the related to #13041, however I even have the files appear without the autosave-feature enabled

Yeah, it's basically the same problem. And it's not related to the auto-saving functionality but to the way we save files in Spyder 4 (I changed the issue title to reflect that).

In order to be sure that saving a file was successful and not corrupting a file in the process, we copy its contents to a temporary file then replace it with the original file.

The problem seems to be that Dropbox retains those files somehow, instead of allowing Spyder to remove them after saving is completed. So for now the only solution is to not use Spyder to work with files in the Dropbox directory.

We'll see what we can do about this for 4.2, to be released in two months or so.

**Comment by ccordoba12** [comment]
I'm also going to close this one as a duplicate of #13041.
